appinventor/components/src/com/google/appinventor/components/common/Direction.java

```diff
@@ -0,0 +1,46 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2020 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.common;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Defines a Direction type used by the Sprite component (and subclasses).
+ */
+public enum Direction implements OptionList {
```

Comment:
Evan has recommended making OptionList generic over the type of underlying value to support better Java typechecking, as in :

    public interface OptionList<T> 
        public T getValue();
    }

I agree with Evan that you should make this change. So the header for Direction should change to 

    public enum Direction implements OptionList<Integer>

And other enum definitions should be similarly changed. 

Note that for new-style completely-abstract enums in which getValue should return the enum itself, you can write something like

    public enum Season implements OptionList<Season>

(I have checked that the self-reference in type in this example works fine in Java.) 



Note that for new-style fully

Reply:
[Done](https://github.com/BeksOmega/appinventor-sources/pull/3/files#diff-6269db0c3cc7a9e6198e281c58dcf275R12)! I also added the ability to [require a static "get" method](https://github.com/BeksOmega/appinventor-sources/pull/3/files#diff-6974a53a12838239a1c808e17a441cd2R1821) with a matching type. Once we finalize the names I can change it :D

Reply:
It is now the finalized name ["fromUnderlyingValue"](https://github.com/BeksOmega/appinventor-sources/pull/3/files#diff-6974a53a12838239a1c808e17a441cd2R1853).

Reply:
Great! Consider this resolved.